### PR TITLE
Oke integration

### PR DESCRIPTION
**Pull request by TomYipOracle**
Second batch of infra code updates from Oracle team.

### CODE REVIEW

File: iam.tf [hunk 1]
```diff
@@ -47,7 +47,7 @@ resource "oci_identity_dynamic_group" "Stack_DG" {
   compartment_id = var.tenancy_ocid
 
   description   = "All Compute instances in stack ${local.Sp3_env_name}"
-  matching_rule = "Any {Any {instance.compartment.id = '${local.Sp3dev_sandbox_cid}'}}"
+  matching_rule = "Any {instance.compartment.id = '${local.Sp3_cid}'}"
```
Comment (TomYipOracle) [inline iam.tf:50]: this fixes a bug where the DG is not including the nodes deployed to the target compartment.

File: oke_cluster_engine.tf [hunk 2]
```diff
@@ -27,6 +27,13 @@ locals {
   is_flexible_node_shape = contains(local.compute_flexible_shapes, var.oke_node_shape)
 }
 
+data "template_file" "oke_node_cloudinit_content" {
```
Comment (TomYipOracle) [inline oke_cluster_engine.tf:30]: This is to allow the OKE nodes to run a command to use all the space assigned to the boot volume, so the pods won't run out of space.

File: oke_cluster_engine.tf [hunk 3]
```diff
@@ -58,10 +65,13 @@ resource oci_containerengine_node_pool oke_containerengine_node_pool {
     }
   }
   node_source_details {
-    boot_volume_size_in_gbs = "100"
+    boot_volume_size_in_gbs = var.oke_boot_vol_size_gb
```
Comment (TomYipOracle) [inline oke_cluster_engine.tf:68]: This allows us to configure the size of boot volume - Bigger volume for prod and load test environments.

File: schema.yaml [hunk 4]
```diff
@@ -385,13 +385,14 @@ variables:
     required: true
     title: "OKE Node Shape"
     description: "Compute shape for nodes in the OKE cluster"
-    default: "VM.Standard2.1"
+    default: "VM.Standard.E3.Flex"
```
Comment (TomYipOracle) [inline schema.yaml:388]: Using moderately powerful shapes for OKE nodes by default.

File: scripts/oke-node.template.sh [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+curl --fail -H "Authorization: Bearer Oracle" -L0 http://169.254.169.254/opc/v2/instance/metadata/oke_init_script | base64 --decode >/var/run/oke-init.sh
+bash /var/run/oke-init.sh
+
+sudo /usr/libexec/oci-growfs -y
```
Comment (TomYipOracle) [inline scripts/oke-node.template.sh:5]: this is the command to make use of all available space in boot volume.

File: schema.yaml (hunk 4 above)
Comment (bhaworth) [reply schema.yaml:388]: Suggest you use E4 not E3.

**Pull request merged**